prog.go

```diff
@@ -307,7 +307,10 @@ func newProgramWithOptions(spec *ProgramSpec, opts ProgramOptions, handles *hand
 		attr.LogSize = uint32(len(logBuf))
 		attr.LogBuf = internal.NewSlicePointer(logBuf)
 
-		_, logErr = internal.BPFProgLoad(attr)
```

Comment:
I think this case is impossible, since we already know that executing the syscall with these arguments must fail. Maybe add a comment to that effect instead of adding the if?

Reply:
In the related issue @ti-mo suggested that a valid fd may be theoretically returned.
Looking at the possible return values from `bpf` syscall, I can think of a transient error condition only if the first call returns `ENOMEM` and the next one succeeds thanks to the OOM freeing memory. Is this possible?
Anyway, even if it were, things would probably be really bad, so I am ok to leave just a comment, if I am not missing something.

Reply:
Looking at it another way: if this case really could happen we would return `nil, nil` from this function which is also wrong.

Reply:
In the freak situation where the second call returns `fd, nil` this code will end up closing the fd and then returning `nil, nil` again?

Reply:
Sorry, I misread the code.